cirq/google/engine/calibration.py

```diff
@@ -174,6 +181,30 @@ def timestamp_str(self,
         dt += datetime.timedelta(microseconds=self.timestamp % 1000000)
         return dt.isoformat(sep=' ', timespec=timespec)
 
+    def str_to_key(self, target: str) -> Union[devices.GridQubit, str]:
+        """Turns a string into a calibration key.
+
+        Attempts to parse it as a GridQubit.  If this fails,
+        returns the string itself.
+        """
+        try:
+            return v2.grid_qubit_from_proto_id(target)
+        except:
+            return target
+
+    def key_to_qubit(self, target: METRIC_KEY) -> devices.GridQubit:
```

Comment:
Similar behavior is used in `noise_models.py` for the values of metrics - would it be worthwhile to have a `value_to_float` function?

Reply:
Wouldn't this just be float(x) though?  I am not sure I see the point in adding this.

The value is Union[str, int, float].  If it is int or float, float() will convert it.  If it is a str, float() will either convert it or raise a ValueError with a descriptive message "could not convert string to float: 'x'"

I think this is exactly what value_to_float would do.  float() is more standard and more concise.

Reply:
`METRIC_VALUE` is a `List[Union[str, int, float]]` - for a value `x`, we use `float(x[0])` to retrieve it. If you don't see a need for this, though, I'm happy to leave as-is.

Reply:
Ok, I added this function to the class.